chain-extensions/dapps-staking/src/lib.rs

```diff
@@ -336,6 +338,28 @@ impl<T: pallet_dapps_staking::Config> ChainExtensionExec<T> for DappsStakingExte
                     Ok(_) => Ok(RetVal::Converging(DSError::Success as u32)),
                 };
             }
+
+            DappsStakingFunc::RebondAndStake => {
+                let contract_bytes: ContractBytes = env.read_as()?;
+                let contract = Self::decode_smart_contract(contract_bytes)?;
+
+                let base_weight =
+                    <T as pallet_dapps_staking::Config>::WeightInfo::rebond_and_stake();
+                env.charge_weight(base_weight)?;
+
+                let caller = env.ext().address().clone();
+                let call_result = pallet_dapps_staking::Pallet::<T>::rebond_and_stake(
+                    RawOrigin::Signed(caller).into(),
+                    contract,
+                );
```

Comment:
Was kinda confused since `rebond_and_stake` is called twice in a row. I do understand that these are two different calls, but still this is a bit confusing. Maybe we should add couple of comments to reduce overall WTF/minute.

Reply:
This can be said for other functions too. it's a more universal topic.
For me it's very clear they are different and what they do because they are different methods, one is `WeightInfo`'s and another is `Pallet`'s.